LEGO1/mxbitmap.cpp

```diff
@@ -0,0 +1,105 @@
+#include "mxbitmap.h"
+#include "mxbool.h"
+#include "mxpalette.h"
+
+// OFFSET: LEGO1 0x100bc980
+MxBitmap::MxBitmap()
+{
+    this->m_info = NULL;
+    this->m_bmiHeader = NULL;
+    this->m_paletteData = NULL;
+    this->m_data = NULL;
+    this->m_unk18 = 0;
+    this->m_palette = NULL;
+}
+
+// OFFSET: LEGO1 0x100bca10
+MxBitmap::~MxBitmap()
+{
+    if (this->m_info != NULL) {
+        free(this->m_info);
+    }
+    if (this->m_data != NULL) {
+        free(this->m_data);
+    }
+    if (this->m_palette != NULL) {
+        // FIXME: Once m_palette/base is understood
+    }
+    // FIXME: Unwind@100bca91. Stack unwinding, I've never done this before.
```

Comment:
This is code generated by the c++ compiler.
The compiler adds code here to first destruct all members with destructors of `MxBitmap`.
Then, it will call the destructor of the parent (=`MxCore`).
```suggestion
```

Hint: in ghidra, you can inline function by enabling the `In Line` function attribute in the function signature editor.

Reply:
The code simply does `delete m_palette;`, which has been compiled into a virtual call to the compiler generated virtual function (which replaces the virtual destructor in the vtable) `MxPalette::'scalar deleting destructor'(unsigned int)`. where the argument `1` tells the generated method to delete the memory. The rest is just generated by the compiler.